source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/utils/hydra.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2022-2024, The Isaac Lab Project Developers.
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Sub-module with utilities for the hydra configuration system."""
+
+
+import functools
+from collections.abc import Callable
+
+import hydra
+from hydra.core.config_store import ConfigStore
+from omegaconf import DictConfig, OmegaConf
+
+from omni.isaac.lab.envs import DirectRLEnvCfg, ManagerBasedRLEnvCfg
+from omni.isaac.lab.utils import replace_slices_with_strings, replace_strings_with_slices
+
+from omni.isaac.lab_tasks.utils.parse_cfg import load_cfg_from_registry
+
+
+def register_task_to_hydra(
+    task_name: str, agent_cfg_entry_point: str
+) -> tuple[ManagerBasedRLEnvCfg | DirectRLEnvCfg, dict]:
+    """Register the task configuration to the Hydra configuration store.
+
+    This function resolves the configuration file for the environment and agent based on the task's name.
+    It then registers the configurations to the Hydra configuration store.
+
+    Args:
+        task_name (str): The name of the task.
+        agent_cfg_entry_point (str): The entry point key to resolve the agent's configuration file.
+
+    Returns:
+        The parsed environment and agent configurations.
+    """
+    # load the configurations
+    env_cfg = load_cfg_from_registry(task_name, "env_cfg_entry_point")
+    agent_cfg = load_cfg_from_registry(task_name, agent_cfg_entry_point)
+    # convert the configs to dictionary
+    env_cfg_dict = env_cfg.to_dict()
+    if isinstance(agent_cfg, dict):
+        agent_cfg_dict = agent_cfg
+    else:
+        agent_cfg_dict = agent_cfg.to_dict()
+    cfg_dict = {"env": env_cfg_dict, "agent": agent_cfg_dict}
+    # replace slices with strings because OmegaConf does not support slices
+    cfg_dict = replace_slices_with_strings(cfg_dict)
+    # store the configuration to Hydra
+    ConfigStore.instance().store(name=task_name, node=cfg_dict)
+    return env_cfg, agent_cfg
+
+
+def hydra_task_config(task_name: str, agent_cfg_entry_point: str) -> Callable:
+    """Decorator to handle the Hydra configuration for a task.
+
+    This decorator registers the task to Hydra and updates the environment and agent configurations from Hydra parsed
+    command line arguments.
+
+    Args:
+        task_name (str): The name of the task.
+        agent_cfg_entry_point (str): The entry point key to resolve the agent's configuration file.
```

Comment:
We only do type-hinting in the signature (as it is easier to maintain).

```suggestion
        task_name: The name of the task.
        agent_cfg_entry_point: The entry point key to resolve the agent's configuration file.
```